application/transaction.go

```diff
@@ -18,21 +22,33 @@ func Transactional(db DB, todo func(f Application) error) error {
 		return errors.WithStack(err)
 	}
 
-	defer func() {
-		switch err {
-		case nil:
-			log.Debug(nil, map[string]interface{}{}, "Commit the transaction!")
-			err = tx.Commit()
-		default:
-			log.Debug(nil, map[string]interface{}{}, "Rolling back the transaction...")
-			_ = tx.Rollback()
-			log.Error(nil, map[string]interface{}{
-				"err": err,
-			}, "database transaction failed!")
-			err = errors.WithStack(err)
+	return func() error {
+		errorChan := make(chan error, 1)
+		txTimeout := time.After(databaseTransactionTimeout)
```

Comment:
ah yes, `time.After(duration)` will initialize a channel and post a message after the given duration. That's  smart ;)
Just to validate the principle, could you add some tests ?